### ISSUE TITLE
client uses wrong server ip address with auto config

### ISSUE THREAD

**Issue by ghost** [input-leap/input-leap#261]
> This issue has been migrated from old Barrier Github repository https://github.com/debauchee/barrier/issues/261
> 
> Issue created on: 2019-03-01 by @ArntWork
> Issue last updated on: 2020-11-28

### Operating Systems ###

Server: Windows 7 SP1

Client: Mac OS 10.13.6

### Barrier Version ###

Server (windows): 2.1.0-RELEASE-0b2dfd80
Client (Mac): 2.1.0-RELEASE-8b69f9fe 

### Steps to reproduce bug ###

Server has multiple network cards and IP addresses due to VM's (using VirtualBox) plus wired and wireless external networks. The 'real' ip addresses (on the external connection) are assigned via dhcp, so they are variable.

The client is set to auto config, and in the log I see tries to connect to an IP address that is internal to the server. This address is the one in bold on the server side, however there is no efficient way to make this the right one. I applied advice from https://symless.com/forums/topic/69-default-bold-ip-address-on-server-is-incorrect-one/ to set the right interface first in the network settings, that did not change the bolded address. I also tried to enter the correct ip address into the 'Interface' field (under networking) but also did not change the bolded address and did not fix the issue.

What is strange is that the only way for the client to get this incorrect IP address, is to have a successful connection with the server, because the address isn't known outside the server...

### Other info ###

When did it occur: after I started the barrier interface on the server (Windows)

Workaround: if I turn off auto-config, but now I have to manually add the ip address on the client, and as said, dhcp, so it changes.

Ideas/suggestions:
- allow me to put a dns name in the current 'server IP' filed on the client This will fix the 'ip address is variable' trouble in my workaround and will generally be an improvement as it means less relying on ip addresses and more on dns names, which is what most of us identify our machine by, I think.
- document the 'Interface' field somehow, so we know what to put there. Or better: make it a dropdown to list interfaces, so we don't have to know what it expects. In case it expects an IP address (that doesn't work for me however), allow a dns name as well
- the ip address that the client wrongly tries to use, comes form the server, so the client already knows how to reach the server? (I don't know how, but whatever it is, it will include the correct ip address as the source I would think?)

Note that https://github.com/debauchee/barrier/issues/81 claims to hold code for a solution, but is closed and I see no way to reopen it. I'm not currently set up to build barrier, unfortunately.

---

> Commented on: 2020-05-20 by @Cobalt6700

I have this exact same issue, using Win 10.  

I have a Hyper-V Virtual Ethernet Adapter and when enabled barrier defaults to this IP range. 

Changing the client side to manual IP allows me to connect - auto connect doesn't work.

---

> Commented on: 2020-11-20 by @enordstrand

Seems to be the same issue I am experiencing. The server has bolded another IP address than the one I want to use, so the Client tries to auto-connect incorrect IP. 

---

> Commented on: 2020-11-24 by @davidthewatson

I've got the same configuration: win10 pro barrier server and macos catalina barrier client. The client ignores the setting in the server IP field and instead goes to another IP address. In the barrier client log on macos, it outputs the following:
```
[2020-11-24T17:45:25 WARNING:  failed to connect to server: Timed out
[2020-11-24T17:45:26] NOTE: connecting to '192.168.50.75 ': 92.242.140.21:24800
```
The correct IP address for the server is 192.168.50.75. The incorrect IP address is 92.242.140.21. This happens after stopping and restarting barrier. I've yet to find a way around it.

Barrier client on linux continues working with the same Win 10 Pro barrier server without issue.

---

> Commented on: 2020-11-28 by @davidthewatson

Turns out that the wrong IP address that I noted above is barefruit in the UK:

https://whatismyipaddress.com/ip/92.242.140.21

Why this breaks the mac barrier client is a mystery. I'm going to take a look at the source and see if I can figure out what's going on because at this point, nothing has changed on my end, but whatever is going on completely broke my mac client in a windows server, linux client, mac client barrier setup that has been working flawlessly for months. 

Not only that, but this error survives clearing barrier's library application state in the local Library directory, and in fact, survives dragging the barrier install to the trash and completely re-installing from the DMG. Whatever is going on, it's persistent.

I may try blocking barefruit at my router because I think the DNS is causing this to go awry and I believe that may be originating with Verizon but no evidence, just a hunch.

---

> Commented on: 2020-11-28 by @davidthewatson

Indeed, I set DNS explicitly at the router and turned off Windows private network firewall on the Windows barrier server, and now my mac barrier client is working again. No barefruit DNS. I'm still not sure why that hijacks the barrier client, but the log I displayed previously:

```
[2020-11-24T17:45:25 WARNING:  failed to connect to server: Timed out
[2020-11-24T17:45:26] NOTE: connecting to '192.168.50.75 ': 92.242.140.21:24800
```

is now:

```
[2020-11-24T17:45:26] NOTE: connecting to '192.168.50.75'
```

without the trailing colon and the IP address to barefruit.

**Comment by pavanyogi** [comment]
yes, this issue still persists. The windows-10 barrier client (v2.4.0-release-3e0d758b) is calculating ip address as 192.168.239.167 while the server displays the ip address as 192.168.239.178. To connect i need to manually add this ip address, which defeats the purpose of auto config.

**Comment by LaraArgento** [comment]
> yes, this issue still persists. The windows-10 barrier client (v2.4.0-release-3e0d758b) is calculating ip address as 192.168.239.167 while the server displays the ip address as 192.168.239.178. To connect i need to manually add this ip address, which defeats the purpose of auto config.

I'm facing the same issue. I keep having to retype the correct ip address, which is frustrating to do multiple times a week with a laptop docked vertically. 
Any ideas?
